src/celeritas/Types.hh

```diff
@@ -43,7 +43,7 @@ using IsotopeId = OpaqueId<struct IsotopeRecord>;
 using MaterialId = OpaqueId<class Material_>;
 
 //! Opaque index of model in the list of physics processes
-using ModelId = OpaqueId<class Model>;
+using ModelId = OpaqueId<class Model_>;
```

Comment:
Why was this needed? There exists a `class Model`... but maybe we should make all of the opaque IDs generic `<struct Foo_>` if don't directly index into an array with element type `T`.

Reply:
It's a forward declaration with a class definition in another namespace: 
```
$ clang-tidy src/celeritas/optical/model/RayleighModel.cc
/home/esseivaj.linux/devel/celeritas/src/celeritas/Types.hh:46:32: error: no definition found for 'Model', but a definition with the same name 'Model' found in another namespace 'celeritas::optical' [bugprone-forward-declaration-namespace,-warnings-as-errors]
   46 | using ModelId = OpaqueId<class Model>;
      |                                ^
/home/esseivaj.linux/devel/celeritas/src/celeritas/optical/model/../Model.hh:29:7: note: a definition of 'Model' is found here
   29 | class Model : public OpticalStepActionInterface, public ConcreteAction
```